### DISCUSSION TITLE
STM32 M4 Clicker

### DISCUSSION THREAD

**Discussion by ITstreet1** [stm32duino/Arduino_Core_STM32#1591] [Variants]
I wonder if some of you tried to make this board work with Arduino IDE.
https://www.mikroe.com/clicker-stm32f4

It has USB HID Bootloader and it shows up as HID-compliant vendor-defined device. While in bootloader mode, when the boot button is pressed, the mode lasts for 5 sec.

There is support for the chip in stm32duino (STM32F415RG). There is an option for HID Bootloader.
When trying to upload the sketch I get the message there is no COM port. Do I have to upload some random sketch first with a serial adapter? If so, what pins should I use? This board doesn't have exposed PA9 and PA10. They are used for other things. They are only USART1 pins.

Or maybe I used the wrong options.

https://download.mikroe.com/documents/starter-boards/clicker/stm32f4/stm32f4-clicker-manual-v100.pdf

**Comment by fpistm** [comment]
i @ITstreet1 
I never tested. Seems you can used STLink.
About HID Bootloader, in a general way you have to have the CDC enabled to be able to have the automatic reset in Bootloader mode.
Maybe the provided HID bootloader is not compatible. I've already saw that with WeAct board.
